packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -225,7 +225,17 @@ describe("SharedTree", () => {
 			onDispose: () => void = () => assert.fail(),
 		): FlexTreeView<TRoot> {
 			const viewSchema = new ViewSchema(defaultSchemaPolicy, {}, schema);
-			return requireSchema(tree.checkout, viewSchema, onDispose, new MockNodeKeyManager());
+			const view = requireSchema(
+				tree.checkout,
+				viewSchema,
+				onDispose,
+				new MockNodeKeyManager(),
+			);
+			const unregister = tree.checkout.storedSchema.on("afterSchemaChange", () => {
```

Comment:
was this required because the tests of the requireSchema function check for disposal or something?

Reply:
The test "differing schema errors and schema change callback" has logic that essentially validates disposal is called on schema upgrade in line 284.